src/riak_kv_compile_tab.erl

```diff
@@ -22,45 +22,112 @@
 
 -module(riak_kv_compile_tab).
 
--export([is_compiling/1]).
--export([get_state/1]).
--export([get_ddl/1]).
--export([new/1]).
--export([insert/4]).
--export([update_state/2]).
+-export([
+         delete_dets/1,
+         get_compiled_ddl_version/1,
+         get_ddl/1,
+         get_state/1,
+         get_ddl_records_needing_recompiling/1,
+         insert/5,
+         is_compiling/1,
+         new/1,
+         update_state/2]).
 
--define(TABLE, ?MODULE).
+-define(LEGACY_TABLE, ?MODULE).
+-define(TABLE, riak_kv_compile_tab_v2).
 
--type compiling_state() :: compiling | compiled | failed.
+-type compiling_state() :: compiling | compiled | failed | retrying.
 -export_type([compiling_state/0]).
 
 -define(is_compiling_state(S),
         (S == compiling orelse
          S == compiled orelse
-         S == failed)).
+         S == failed orelse
+         S == retrying)).
 
 %% 
 -spec new(file:name()) ->
-        {ok, dets:tab_name()} | {error, any()}.
+         {ok, dets:tab_name(), dets:tab_name()} | {error, any(), any()}.
 new(FileDir) ->
+    LegacyFilePath = filename:join(FileDir, [?LEGACY_TABLE, ".dets"]),
+    lager:debug("Opening legacy DDL DETS table ~s", [LegacyFilePath]),
     FilePath = filename:join(FileDir, [?TABLE, ".dets"]),
-    dets:open_file(?TABLE, [{type, set}, {repair, force}, {file, FilePath}]).
+    lager:debug("Opening DDL DETS table ~s", [FilePath]),
+    Result1 = dets:open_file(?LEGACY_TABLE, [{type, set}, {repair, force}, {file, LegacyFilePath}]),
+    Result2 = dets:open_file(?TABLE, [{type, set}, {repair, force}, {file, FilePath}]),
+    new_table_result(Result1, Result2).
+
+%% Open both old and new copies of the DETS table.  Assume old table is
+%% canonical and force entries in the new table to be recompiled.
+-spec new_table_result({ok, dets:tab_name()} | {error, any()},
+                       {ok, dets:tab_name()} | {error, any()}) ->
+                       {ok, dets:tab_name(), dets:tab_name()} | {error, any(), any()}.
+new_table_result({ok, LegacyDets}, {ok, Dets}) ->
+    %% Convert all legacy DDL compilation records to current format
+    upgrade_legacy_records(),
+
+    %% TODO: Use this once 1.2 support has been removed
```

Comment:
This TODO looks out of place and is confusing. I assume you meant something above the `upgrade_legacy_records` invocation, and a verb other than "Use"